### PR TITLE
feat: adding first pass of get services by id

### PR DESCRIPTION
**Pull request by lgarofalo**

### CODE REVIEW

**Comment by lgarofalo** [comment]
This is not feature complete yet, but the first pass of the direct fields
```
{
  "service": {
    "updated_at": "2023-08-03 06:32:29.490801 +0000 +0000",
    "alternate_name": "",
    "application_process": "",
    "certified": false,
    "eligibility": "",
    "email": "",
    "fee": "",
    "id": 1,
    "interpretation_services": "",
    "long_description": "Call for an appt. M-F 8am-4:30pm for psychiatric services, medical triage, support groups, informational classes, medical and HIV primary care, podiatry, substance abuse treatment & housing referrals. Drop-in for primary care M-F 8am. Drop-in social work clinic M-F 8am-12pm. Sobriety Support group MWF 9:30am. Harm Reduction Group Tu 9:30-10am. AA meetings Th 6:30-7:30pm. Call for schedule of free shuttle to Ft. Miley.\n",
    "name": "Legros-Lynch",
    "required_documents": "",
    "url": "",
    "verified_at": "0001-01-01 00:00:00 +0000 UTC",
    "wait_time": "",
    "certified_at": "0001-01-01 00:00:00 +0000 UTC",
    "featured": false,
    "source_attribution": 0,
    "status": 1,
    "internal_note": "",
    "short_description": "temp"
  }
}
```

I need to investigate why short description wasn't available in my DB, clean up testing, and add data from the other objects for 

schedule
notes
categories
addresses
eligibilities
instructions
documents
resource
program

File: internal/db/types.go [hunk 1]
```diff
@@ -24,8 +24,31 @@ type ChangeRequest struct {
 }
 
 type Service struct {
-	Id         int
-	ResourceId int
+	Id                     int
+	CreatedAt              sql.NullTime
+	UpdatedAt              sql.NullTime
```
Comment (richardxia) [inline internal/db/types.go:29]: I'm not sure if this Go sql package has a non-nullable Time type, but according to the [schema.rb](https://github.com/ShelterTechSF/askdarcel-api/blob/8559a6e704c604c3fb1285a299ccacc03a706fd6/db/schema.rb#L382-L383) file in the Rails app, created_at and updated_at are not nullable

File: internal/db/types.go [hunk 2]
```diff
@@ -24,8 +24,31 @@ type ChangeRequest struct {
...
+	LongDescription        sql.NullString
+	Eligibility            sql.NullString
+	RequiredDocuments      sql.NullString
+	Fee                    sql.NullString
+	ApplicationProcess     sql.NullString
+	ResourceId             sql.NullInt32
+	VerifiedAt             sql.NullTime
+	Email                  sql.NullString
+	Status                 sql.NullInt32
+	Certified              sql.NullBool
```
Comment (richardxia) [inline internal/db/types.go:40]: This certified field is also not nullable according to the Rails schema.rb file.

File: internal/services/types.go [hunk 3]
```diff
@@ -0,0 +1,28 @@
+package services
+
+type ServiceResponse struct {
+	Service *Service `json:"service"`
+}
+type Service struct {
+	UpdatedAt              string `json:"updated_at"`
+	AlternateName          string `json:"alternate_name"`
```
Comment (richardxia) [inline internal/services/types.go:8]: My comments on the SQL null thing are making me wonder about the nullability of fields in this struct. Does the way you've declared these types allow any of these to be null? What happens if the field is null in the database when you unmarshal into this struct?

Also, if it's the case that this code will return the empty string when in encounters a null in the database, I'm not 100% sure if that is drop-in equivalent for our frontend code. We fortunately converted all the frontend code to TypeScript, and we explicitly have `nulls` in the frontend's version of this struct (https://github.com/ShelterTechSF/askdarcel-web/blob/70664042421e2278e34d9b2c2822c827fb625eeb/app/models/Service.ts#L20-L52). In JavaScript, the empty string, `null`, and `undefined` will fortunately evaluate to `false` inside of conditional expressions, but we'd have to double check that we're not doing explicit equality checks against `null` anywhere in the frontend.

File: internal/db/types.go (hunk 2 above)
Comment (lgarofalo) [reply internal/db/types.go:40]: Yea I'll do a pass to make sure anything required in DB is expected here.

File: internal/services/types.go (hunk 3 above)
Comment (lgarofalo) [reply internal/services/types.go:20]: I will match this so these can return null in the same manner we do currently, if we want to tweak this in the future we can but not a problem to change for now.

Comment (lgarofalo) [reply internal/services/types.go]: added a commit that makes this adjustment

**Comment by lgarofalo** [comment]
This PR is getting large and I want to test this out, it's not complete yet but a lot of progress in here so I'll cut a release and see how it looks in prod/staging

**Comment by lgarofalo** [comment]
Current state
```
{
  "service": {
    "id": 5,
    "alternate_name": null,
    "application_process": null,
    "certified": false,
    "eligibility": null,
    "email": null,
    "fee": null,
    "interpretation_services": null,
    "long_description": "Six (6) services per week in San Francisco.\nTue:\n8:30am - 2pm\nWed:\n8:30am - 12pm, 2pm - 5pm\nThu - Fri:\n8:30am - 2pm\nSat:\n7am - 1pm\n",
    "name": "Beer-Windler",
    "required_documents": null,
    "url": null,
    "verified_at": null,
    "wait_time": null,
    "certified_at": null,
    "featured": null,
    "source_attribution": 0,
    "status": 1,
    "internal_note": null,
    "short_description": "temp",
    "updated_at": "2023-08-03 06:32:29.696282 +0000 +0000",
    "resource": {
      "id": 3,
      "alternate_name": null,
      "certified": false,
      "email": null,
      "legal_status": null,
      "long_description": "Urgent and Primary Care, Dental, Podiatry, TB, STD, HIV, and Pregnancy testing, Health for Women, Family Planning, Prenatal Care, and Immunizations. Sliding scale (outpatient medical services free to homeless). Call for appt. Pediatrics & optometry available.\n",
      "name": "Abbott-Reinger",
      "short_description": "Culpa ut nemo provident.",
      "status": "1",
      "verified_at": null,
      "website": "http://schmidt-padberg.net/stanford.olson",
      "certified_at": null,
      "featured": null,
      "source_attribution": "0",
      "internal_note": null
    },
    "program": null,
    "Schedule": {
      "id": 8,
      "hours_known": true,
      "schedule_days": []
    },
    "notes": [
      {
        "id": 8,
        "note": "Enim adipisci sunt. Autem iste veritatis. Nesciunt est voluptas."
      }
    ],
    "categories": [
      {
        "id": 68,
        "name": "Prevent & Treat",
        "top_level": false,
        "featured": false
      },
      {
        "id": 159,
        "name": "Homelessness Essentials",
        "top_level": true,
        "featured": false
      }
    ],
    "addresses": [],
    "eligibilities": [],
    "instructions": [
      {
        "id": 5,
        "instruction": "Instruction text goes here"
      }
    ],
    "documents": [
      {
        "id": 5,
        "name": "5",
        "url": "document_url",
        "description": "Some document"
      }
    ]
  }
}
```

**Pull request merged**